Homework4/src/ro/siit/homework4/Store.java

```diff
@@ -0,0 +1,73 @@
+package ro.siit.homework4;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Store {
+
+    static Map<String, StockItem> stock;
+    static List<SoldItem> sold;
+
+    @Override
+    public String toString() {
+        return "Store{" +
+                "stock=" + stock +
+                ", sold=" + sold +
+                '}';
+    }
+
+    public Store() {
+        this.stock = new HashMap<>();
+        this.sold = new ArrayList<>();
+    }
+
+    public static void add(StockItem product) {
+
+        stock.forEach((k, v) -> {
+
+            if (v.getProduct().getName() == product.getProduct().getName()) {
+                int tQ = v.getQuantity() + product.getQuantity();
+                v.setQuantity(tQ);
+            }
+        });
```

Comment:
so new product types cannot be added, you can only add supported products. not ideal